### PR TITLE
impl: decrease allocation in string to_literal

### PR DESCRIPTION
**Pull request by mfp**
- faster + keeps 4.02 compat
- breaks the API but it was barely made public? (add alias to `String.split_on_char` and abandon 4.02 if it matters)

### CODE REVIEW

File: impl/ocaml/sqlgg_runtime.ml [hunk 1]
```diff
@@ -1,10 +1,32 @@
...
+  Buffer.contents b
+
+(* cf. https://sqlite.org/lang_expr.html "Literal Values"
+    "A string constant is formed by enclosing the string in single quotes
+    ('). A single quote within the string can be encoded by putting two
+    single quotes in a row - as in Pascal. C-style escapes using the
+    backslash character are not supported because they are not standard
+    SQL." *)
+let sqlite3_string_to_literal s =
+  let b = Buffer.create (String.length s + String.length s lsl 2) in
```
Comment (ygrek) [inline impl/ocaml/sqlgg_runtime.ml:24]: > lsl 2

lets not go overboard, `* 4` is easier to read. But why such a huge buffer? it will never get fully used

**Review by ygrek — CHANGES_REQUESTED** [review]
breaking api doesn't matter, it was not part of release yet
the only qn is if these funcs should just be in traits? duplicated for mysql/mariadb but that's fine?

File: impl/ocaml/sqlgg_runtime.ml (hunk 1 above)
Comment (mfp) [reply impl/ocaml/sqlgg_runtime.ml]: that was me trolling for the laughs with the bit shifting and shooting myself in the foot; changed to `len + len / 4`

**Comment by mfp** [comment]
Will move the functions tomorrow.

**Review by ygrek — APPROVED** [review]

**Pull request merged**